earth_enterprise/src/fusion/autoingest/plugins.src/PacketGen.src

```diff
@@ -111,17 +130,42 @@ void PacketGenAssetVersionImplD::UpdateChildren(
     const khLevelCoverage levCov(config.coverage.levelCoverage(level));
     std::vector<uint> neededIndexes;
 
+    // Filter based on quadtree path
+    khExtents<double> doubleExtents = levCov.degExtents(tilespace);
+    int tempLevel = 0;
+    QuadtreePath coverageMBR = tilespaceIndex.getQuadtreeMBR(
+        doubleExtents,
+        tempLevel,
+        config.endMinifyLevel);
+    if (coverageMBR != prevCoverageMBR || qualifiedInsetInfoPtrs.size() == 0) {
+      qualifiedInsetInfoPtrs.clear();
+      qualifiedInsetInfoPtrs.reserve(extra.insetInfo.size());
+      qualifiedInsetInfoPtrs = 
+          std::move(tilespaceIndex.intersectingExtents(
+              coverageMBR, 
+              config.beginMinifyLevel, 
+              config.endMinifyLevel));
+      prevCoverageMBR = coverageMBR;
+    }
+
     neededIndexes.reserve(numInsets);
     auto gencov = khInsetCoverage(levCov);
     overlapEnvelope<ProductAssetVersion> env(
         type,
-        extra.insetInfo,
-        numInsets,
+        qualifiedInsetInfoPtrs,
+        qualifiedInsetInfoPtrs.size(),
         config.beginMinifyLevel,
         config.endMinifyLevel,
         level);
     extra.calc->setEnvelope(env);
-    neededIndexes = std::move(extra.calc->GetOverlapForLevel(gencov));
+    std::vector<uint> tempNeededIndexes = std::move(extra.calc->GetOverlapForLevel(gencov));
```

Comment:
Minor: I think you could save some time here by storing the index directly in the QuadKeyTree rather than a pointer to the inset info and then using a separate map to look up the index. I haven't thought it through completely, but if we can make that work it would improve performance.

Reply:
I'm not sure if you mean to store the index *instead of* the inset pointer or in addition to. There's probably a bit of a wrinkle either way since we need a vector of `InsetInfo*` to pass into `GetOverlapForLevel`. Either way there would be some copying involved, I think.
But to your point, this is pretty brute force, and there's almost certainly a way to make it more efficient.

Reply:
Yeah, that's basically what I'm getting at - this is a place where we can optimize.